### PR TITLE
Improve user settings

### PR DESCRIPTION
**Pull request by qwerty287**
Make the user settings tabbed as well, currently only holding "Settings" (language) and "API". Can be extended with user secrets (#1739) etc. later.
Screenshot 2023-08-02 at 09-56-00 Woodpecker
Screenshot 2023-08-02 at 09-56-05 Woodpecker

### CODE REVIEW

**Comment by xoxys** [comment]
Thanks! I had the same fix on my list.

File: web/src/views/org/OrgRepos.vue [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     <template #titleActions>
       <IconButton
-        v-if="orgPermissions.admin"
+        v-if="!org.is_user && orgPermissions.admin"
```
Comment (xoxys) [inline web/src/views/org/OrgRepos.vue:9]: What's the purpose of the changed permission check to the org views?

File: web/src/views/admin/Admin.vue [hunk 2]
```diff
@@ -1,24 +0,0 @@
-<template>
```
Comment (xoxys) [inline web/src/views/admin/Admin.vue:1]: Is the change to the admin view and router somehow related to the restructured user settings, or just a "drive-by" cleanup?

File: web/src/views/org/OrgRepos.vue (hunk 1 above)
Comment (xoxys) [reply web/src/views/org/OrgRepos.vue:9]: Im a bit confused. AFAIK Woodpecker threads users as orgs. If I look at a repo from my personal user space, this is handled as

Comment (xoxys) [reply web/src/views/org/OrgRepos.vue:9]: Thank you for the explanation. Hmm, as you wrote in https://github.com/woodpecker-ci/woodpecker/issues/1739 the backend is the same and as we handle users as orgs now this should IMO just work? Or what's technically missing that there are no user secrets yet? Just removing it because we might want to move user secrets to another location sounds bad to me. But I'm not that deep into the codebase yet either, sorry if the questions are bullshit :see_no_evil:

Comment (xoxys) [reply web/src/views/org/OrgRepos.vue:9]: Thanks :+1: works for me as well.

File: web/src/assets/locales/en.json [hunk 3]
```diff
@@ -425,16 +425,24 @@
         }
     },
     "user": {
-        "settings": "User Settings",
+        "user_settings": "User Settings",
```
[the remark this answers is not in the document]
Comment (xoxys) [reply web/src/assets/locales/en.json]: Can we not just use the same structure we have for the admin settings already?

```
   "user":{
      "settings":{
         "settings":"User Settings",
         "oauth_error":"Error while authenticating against OAuth provider",
         "internal_error":"Some internal error occurred",
         "access_denied":"You are not allowed to login",
         "general":{
            "general":"General",
            "language":"Language"
         },
         "api":{
            "api":"API",
            "token":"Personal Access Token",
            "shell_setup":"Shell setup",
            "api_usage":"Example API Usage",
            "cli_usage":"Example CLI Usage",
            "dl_cli":"Download CLI",
            "shell_setup_before":"do shell setup steps before",
            "reset_token":"Reset token",
            "swagger_ui":"Sagger UI"
         }
      }
   }
```

File: web/components.d.ts [hunk 4]
```diff
@@ -101,6 +101,9 @@ declare module '@vue/runtime-core' {
     Tab: typeof import('./src/components/layout/scaffold/Tab.vue')['default']
     Tabs: typeof import('./src/components/layout/scaffold/Tabs.vue')['default']
     TextField: typeof import('./src/components/form/TextField.vue')['default']
+    UserAPITab: typeof import('./src/components/user/UserAPITab.vue')['default']
+    UserGeneralTab: typeof import('./src/components/user/UserGeneralTab.vue')['default']
+    UserSettingsTab: typeof import('./src/components/user/UserSettingsTab.vue')['default']
```
Comment (xoxys) [inline web/components.d.ts:106]: Was renamed and can be removed?

**Review by xoxys — CHANGES_REQUESTED** [review]

**Comment by xoxys** [comment]
What do you think about adding a headline to the tabs, as we have in the admin settings?
image

**Comment by xoxys** [comment]
Or repo settings without a subtitle (which is not required if we don't really have to explain something)

image

File: web/src/components/user/UserAPITab.vue [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      </div>
+    </div>
+
+    <div class="mt-2 mb-4">
+      <div class="flex items-center mb-2">
+        <div class="flex items-center text-wp-text-100 font-bold">
+          <label>{{ $t('user.settings.api.token') }}</label>
+        </div>
+        <Button class="ml-4" :text="$t('user.settings.api.reset_token')" @click="resetToken" />
+      </div>
```
Comment (xoxys) [inline web/src/components/user/UserAPITab.vue:16]:
```suggestion
      <div class="flex items-center mb-2">
        <div class="flex flex-row items-center text-wp-text-100 font-bold">
          <label>{{ $t('user.settings.api.token') }}</label>
        </div>
        <Button class="ml-auto" :text="$t('user.settings.api.reset_token')" @click="resetToken" />
      </div>
```

What do you think about aligning the button to the right?

image

**Review by xoxys — APPROVED** [review]
I like it :heart:

File: web/src/assets/locales/en.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (lafriks) [reply web/src/assets/locales/en.json]: yes, lgtm except errors are not related to settings but are user related so these should not be moved into settings section
